source/store/AppContext.tsx

```diff
@@ -0,0 +1,62 @@
+import PropTypes from "prop-types";
+import React, { useState, useEffect } from "react";
+import env from "react-native-config";
+import ConfigurationService from "../services/ConfigurationService";
+import StorageService, { APP_ENV_KEY } from "../services/StorageService";
+
+const AppContext = React.createContext({});
+
+export const Configuration = new ConfigurationService(
```

Comment:
I dont think that the ConfigurationService should be instantiated in the AppContext at all. This should preferably be instantiated in App.tsx file which is the entrypoint of the application.

And also, perhaps rename the configuration to environmentConfiguration. The name configuration may not describe exactly which configurations we are talking about.